web-authentication-ui/src/main/java/com/okta/webauthenticationui/WebAuthenticationClient.kt

```diff
@@ -40,31 +38,40 @@ import com.okta.webauthenticationui.events.CustomizeCustomTabsEvent
  * To further customize the authentication flow, please read more about the underlying flows: [AuthorizationCodeFlow],
  * [RedirectEndSessionFlow].
  */
-class WebAuthenticationClient private constructor(
+class WebAuthenticationClient(
     private val oidcClient: OidcClient,
-    private val webAuthenticationProvider: WebAuthenticationProvider,
+    private val webAuthenticationProvider: WebAuthenticationProvider = DefaultWebAuthenticationProvider(oidcClient.configuration.eventCoordinator),
 ) {
     companion object {
         init {
             SdkVersionsRegistry.register(SDK_VERSION)
         }
-
-        /**
-         * Initializes a web authentication client using the [OidcClient].
-         *
-         * @receiver the [OidcClient] used to perform the low level OIDC requests, as well as with which to use the configuration from.
-         * @param webAuthenticationProvider the [WebAuthenticationProvider] which will be used to show the UI when performing the
-         * redirect flows.
-         */
-        fun OidcClient.createWebAuthenticationClient(
-            webAuthenticationProvider: WebAuthenticationProvider = DefaultWebAuthenticationProvider(configuration.eventCoordinator)
-        ): WebAuthenticationClient {
-            return WebAuthenticationClient(this, webAuthenticationProvider)
-        }
     }
 
-    private val authorizationCodeFlow: AuthorizationCodeFlow = oidcClient.createAuthorizationCodeFlow()
-    private val redirectEndSessionFlow: RedirectEndSessionFlow = oidcClient.createRedirectEndSessionFlow()
+    /**
+     * Initializes a web authentication client.
+     *
+     * @param webAuthenticationProvider the [WebAuthenticationProvider] which will be used to show the UI when performing the
+     * redirect flows.
+     */
+    constructor(
+        webAuthenticationProvider: WebAuthenticationProvider = DefaultWebAuthenticationProvider(OidcConfiguration.default.eventCoordinator)
+    ) : this(OidcClient.default, webAuthenticationProvider)
+
+    /**
+     * Initializes a web authentication client.
+     *
+     * @param oidcConfiguration the [OidcConfiguration] specifying the authorization servers.
+     * @param webAuthenticationProvider the [WebAuthenticationProvider] which will be used to show the UI when performing the
+     * redirect flows.
+     */
+    constructor(
+        oidcConfiguration: OidcConfiguration,
+        webAuthenticationProvider: WebAuthenticationProvider = DefaultWebAuthenticationProvider(oidcConfiguration.eventCoordinator)
+    ) : this(OidcClient.createFromConfiguration(oidcConfiguration), webAuthenticationProvider)
+
+    private val authorizationCodeFlow: AuthorizationCodeFlow = AuthorizationCodeFlow(oidcClient)
+    private val redirectEndSessionFlow: RedirectEndSessionFlow = RedirectEndSessionFlow(oidcClient)
```

Comment:
You might want to make the getter for these public, so a developer can hook into the event lifecycle / customize the behavior of the underlying flows.  This allows a developer to use the higher-level WebAuthentication class, while still being able to tweak / monitor the underlying details, without having to rebuild everything from scratch.